worker/src/handlers/download.rs

```diff
@@ -104,6 +104,14 @@ pub async fn process(job_id: Uuid, payload: JobMessage) -> Result<DownloadResult
         total_bytes += chunk_size;
 
         let current_time = Utc::now();
+        let elapsed_time = current_time - download_start_time;
+        if elapsed_time >= MAX_DOWNLOAD_DURATION {
```

Comment:
While probably enough for first version, lets make a Jira issue to monitor this from a separate tokio task.
If we monitor it from inside, we risk the server stopping sending data, or sending it extremely slowly, which will block us in `response.chunk().await` perhaps indefinitely and we won't exit on timeout as we should.

Reply:
Actually, maybe just [enforcing some timeout](https://docs.rs/tokio/latest/tokio/time/fn.timeout.html) on `response.chunk().await` would work? Each chunk could have a timeout equal to `MAX_DOWNLOAD_DURATION - elapsed_time` or smth.

Reply:
@kacperzuk-neti I've implemented the `tokio::time::timeout` but its not ideal, in worst case the job will take 2x `MAX_DOWNLOAD_DURATION` time